src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiNotificationBadge,
+  EuiPopover,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { SavedObjectsClientContract, ToastsStart } from 'opensearch-dashboards/public';
+import { getDataSourcesWithFields } from '../utils';
+import { SavedObject } from '../../../../../core/public';
+import { DataSourceAttributes } from '../../types';
+
+interface DataSourceAggregatedViewProps {
+  savedObjectsClient: SavedObjectsClientContract;
+  notifications: ToastsStart;
+  hideLocalCluster: boolean;
+  fullWidth: boolean;
+  activeDataSourceIds?: string[];
+  dataSourceFilter?: (dataSource: SavedObject<DataSourceAttributes>) => boolean;
+  displayAllCompatibleDataSources: boolean;
+}
+
+interface DataSourceAggregatedViewState {
+  isPopoverOpen: boolean;
+  allDataSourcesIdToTitleMap: Map<string, any>;
+}
+
+export class DataSourceAggregatedView extends React.Component<
+  DataSourceAggregatedViewProps,
+  DataSourceAggregatedViewState
+> {
+  private _isMounted: boolean = false;
+
+  constructor(props: DataSourceAggregatedViewProps) {
+    super(props);
+
+    this.state = {
+      isPopoverOpen: false,
+      allDataSourcesIdToTitleMap: new Map(),
+    };
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  onClick() {
+    this.setState({ ...this.state, isPopoverOpen: !this.state.isPopoverOpen });
+  }
+
+  closePopover() {
+    this.setState({ ...this.state, isPopoverOpen: false });
+  }
+
+  async componentDidMount() {
+    this._isMounted = true;
+    getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type'])
+      .then((fetchedDataSources) => {
+        if (fetchedDataSources?.length) {
+          let filteredDataSources = fetchedDataSources;
+          if (this.props.dataSourceFilter) {
+            filteredDataSources = fetchedDataSources.filter((ds) =>
+              this.props.dataSourceFilter!(ds)
+            );
+          }
+
+          const allDataSourcesIdToTitleMap = new Map();
+
+          filteredDataSources.forEach((ds) => {
+            allDataSourcesIdToTitleMap.set(ds.id, ds.attributes!.title || '');
+          });
+
+          if (!this.props.hideLocalCluster) {
+            allDataSourcesIdToTitleMap.set('', 'Local cluster');
+          }
+
+          if (!this._isMounted) return;
+          this.setState({
+            ...this.state,
+            allDataSourcesIdToTitleMap,
+          });
+        }
+      })
+      .catch(() => {
+        this.props.notifications.addWarning(
```

Comment:
We catch warning notifications here, but we didn't change the component state, do we need to set state to reflect the error condition?

Reply:
If there is error, this will show a toast about the error